### PR TITLE
Convert BsSpinner to TypeScript

### PR DESCRIPTION
**Pull request by Baltazore**
Converts <BsSpinner> to TypeScript. Including the tests.

### CODE REVIEW

**Comment by Baltazore** [comment]
@jelhan thanks for review, updated.

**Review by jelhan — APPROVED** [review]
Looks good to me. 👍

Will give Simon and Sander the chance to review as well before hitting the merge button. In case there isn't an update in next 2-3 days, please ping me. I may forget about it without notification.

File: pnpm-lock.yaml
Comment (SanderKnauff) [inline pnpm-lock.yaml:1]: Huh, I wonder why this was updated since the rest of the lockfile seems the same...

**Review by SanderKnauff — APPROVED** [review]
Looks good to me! I have just a minor uncertainty about the lockfile change, but that might just have been pnpm updating that.

Since CI is fine with it, it probably is not that big of an issue.

File: pnpm-lock.yaml
Comment (jelhan) [reply pnpm-lock.yaml:1]: Good catch. I ignored that.

PNPM updates the lockfile version when running `pnpm install`. But `6.1` is a value, which we don't want. That's from an old PNPM version and was rolled back. See https://github.com/orgs/pnpm/discussions/6857.

Let's reset that change:

```suggestion
lockfileVersion: '6.0'
```

@Baltazore: Looks as if you need to upgrade PNPM on your machine.

Comment (Baltazore) [reply pnpm-lock.yaml]: yeah I had pnpm 8.6 locked due to ember-data, sorry ...

**Pull request merged**
